src/pages/settings/PaymentsPage.js

```diff
@@ -1,21 +1,110 @@
 import React from 'react';
+import {Pressable, TextInput, View} from 'react-native';
+import PropTypes from 'prop-types';
+import {withOnyx} from 'react-native-onyx';
+
+import CONST from '../../CONST';
+import ONYXKEYS from '../../ONYXKEYS';
+import ROUTES from '../../ROUTES';
+import HeaderGap from '../../components/HeaderGap';
 import HeaderWithCloseButton from '../../components/HeaderWithCloseButton';
+import Text from '../../components/Text';
 import {redirect} from '../../libs/actions/App';
-import HeaderGap from '../../components/HeaderGap';
-import ROUTES from '../../ROUTES';
+import NameValuePair from '../../libs/actions/NameValuePair';
+import {fetch} from '../../libs/actions/User';
+import styles from '../../styles/styles';
+
+const propTypes = {
+    // The PaypalMe Username
+    paypalMeAddress: PropTypes.string,
+};
+
+const defaultProps = {
+    paypalMeAddress: '',
+};
+
+class PaymentsPage extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            paypalMeAddress: props.paypalMeAddress,
+        };
+        this.setPaypalAccount = this.setPaypalAccount.bind(this);
+    }
+
+    componentDidMount() {
+        fetch();
+    }
+
+    componentDidUpdate(prevProps) {
+        if (prevProps.paypalMeAddress !== this.props.paypalMeAddress) {
+            // eslint-disable-next-line react/no-did-update-set-state
+            this.setState({paypalMeAddress: this.props.paypalMeAddress});
+        }
+    }
+
+    /**
+     * Sets the paypalMeAddress for the current user
+     */
+    setPaypalAccount() {
+        NameValuePair.set(CONST.NVP.PAYPAL_ME_ADDRESS, this.state.paypalMeAddress, ONYXKEYS.NVP_PAYPAL_ME_ADDRESS);
+    }
 
-const PaymentsPage = () => (
-    <>
-        <HeaderGap />
-        <HeaderWithCloseButton
-            title="Payments"
-            shouldShowBackButton
-            onBackButtonPress={() => redirect(ROUTES.SETTINGS)}
-            onCloseButtonPress={() => redirect(ROUTES.HOME)}
-        />
-    </>
-);
+    render() {
+        return (
+            <>
+                <HeaderGap />
+                <HeaderWithCloseButton
+                    title="Payments"
+                    shouldShowBackButton
+                    onBackButtonPress={() => redirect(ROUTES.SETTINGS)}
+                    onCloseButtonPress={() => redirect(ROUTES.HOME)}
+                />
+                <View style={[styles.flex1, styles.pageWrapper]}>
```

Comment:
Do we really need all of these nested `<Views>`s? I am also curious why we need `styles.pageWrapper` - we already have a utility style for padding. Perhaps we can get rid of pageWrapper aand the `styles.pv3` below and just add something like `styles.p5` here?